// react-test-renderer/lib/shallow/ReactTestTextComponent.js
/**
 * Copyright 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * 
 */

'use strict';

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var ReactTestTextComponent = function () {
  function ReactTestTextComponent(element) {
    _classCallCheck(this, ReactTestTextComponent);

    this._currentElement = element;
  }

  ReactTestTextComponent.prototype.receiveComponent = function receiveComponent(nextElement) {
    this._currentElement = nextElement;
  };

  ReactTestTextComponent.prototype.toJSON = function toJSON() {
    return this._currentElement;
  };

  ReactTestTextComponent.prototype.mountComponent = function mountComponent() {};

  ReactTestTextComponent.prototype.getHostNode = function getHostNode() {};

  ReactTestTextComponent.prototype.unmountComponent = function unmountComponent() {};

  return ReactTestTextComponent;
}();

module.exports = ReactTestTextComponent;
